File: competitive/AtCoder/ABC105/D.py
# TLE

import numpy as np

line = input().split()
N = int(line[0])
M = int(line[1])
line = input().split()
A = [0]

A.extend([int(l) for l in line])

numA = np.array(A, np.int64)
cumA = np.cumsum(numA)
length = cumA.shape[0]

num = 0
for i in range(length):
    for j in range(i+1, length):
        if (cumA[j] - cumA[i]) % M == 0:
            num += 1
print(num)

File: competitive/AtCoder/ABC105/D.cpp
// TLE

#include <iostream>
using namespace std;

int main()
{
    long N, M;

    cin >> N;
    cin >> M;

    long long *cumA = new long long[N + 1];

    long long A;
    long long cum = 0;
    cumA[0] = 0;

    for (int i = 1; i < N+1; ++i)
    {
        cin >> A;
        cum += A;
        cumA[i] = cum;
    }
  
    int num = 0;

    for (int i = 0; i < N+1; ++i)
    {
        for (int j = i+1; j < N+1; ++j)
        {
            if ((cumA[j] - cumA[i]) % M == 0)
            {
                num += 1;
            }
        }
    }

    cout << num << endl;
    return 0;
}
